src/encoded/static/components/search.js

```diff
@@ -711,30 +711,48 @@ TypeTerm.propTypes = {
     total: PropTypes.number.isRequired,
 };
 
-
 class Facet extends React.Component {
-    constructor() {
-        super();
+    constructor(props) {
+        super(props);
 
         // Set initial React commponent state.
         this.state = {
-            facetOpen: false,
+            filteredTerms: null,
         };
 
         // Bind `this` to non-React methods.
-        this.handleClick = this.handleClick.bind(this);
+        this.handleSearch = this.handleSearch.bind(this);
     }
 
-    handleClick() {
-        this.setState(prevState => ({ facetOpen: !prevState.facetOpen }));
+    handleSearch(event) {
+        // search term entered by the user
+        const filterVal = String(event.target.value.toLowerCase().replace(/ /g, '').replace(/[^\w\s]/gi, ''));
+
+        // which facet terms match the search term entered by the user
+        let terms = this.props.facet.terms.filter((term) => {
+            if (term.doc_count > 0) {
+                const termKey = term.key.toLowerCase().replace(/ /g, '').replace(/[^\w\s]/gi, '');
```

Comment:
Since this is the same transform as you do above, maybe make a one-line stand-alone arrow function private to search.js that does this transformation? Just to avoid having to make the same change in two different places in the future.